skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,381 @@
+        #                   |     (see scipy.stats.lognorm)
+        # Laplace           | loc = mean, scale = scale parameter
+
+        dist_params = {
+            "Normal": ["loc", "scale"],
+            "Laplace": ["loc", "scale"],
+            "TDistribution": ["loc", "scale"],
+            "Poisson": ["mu"],
+            "LogNormal": ["scale", "s"],
+        }
+
+        skpro_params = {
+            "Normal": ["mu", "sigma"],
+            "Laplace": ["mu", "scale"],
+            "TDistribution": ["mu", "sigma"],
+            "Poisson": ["mu"],
+            "LogNormal": ["mu", "sigma"],
+        }
+
+        kwargs = {}
+
+        if self.dist in dist_params and self.dist in skpro_params:
+            ngboost_params = dist_params[self.dist]
+            skp_params = skpro_params[self.dist]
+            for ngboost_param, skp_param in zip(ngboost_params, skp_params):
+                kwargs[skp_param] = self._pred_dist(X).params[ngboost_param]
+                if self.dist == "LogNormal" and ngboost_param == "scale":
+                    kwargs[skp_param] = np.log(self._pred_dist(X).params[ngboost_param])
+
+                kwargs[skp_param] = self._check_y(y=kwargs[skp_param])
+                # returns a tuple so taking only first index of the tuple
+                kwargs[skp_param] = kwargs[skp_param][0]
+            kwargs["index"] = X.index
+            kwargs["columns"] = self._y_cols
+
+        # Convert NGBoost Distribution to skpro BaseDistribution
+        pred_dist = self._ngb_dist_to_skpro(**kwargs)
+
+        return pred_dist
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params1 = {"dist": "Normal"}
+        params2 = {
+            "dist": "Laplace",
+            "n_estimators": 800,
+        }
+        params3 = {}
+        params4 = {
+            "dist": "Poisson",
+            "minibatch_frac": 0.8,
+            "early_stopping_rounds": 4,
+        }
+        params5 = {
+            "dist": "LogNormal",
+            "learning_rate": 0.001,
+            "validation_fraction": 0.2,
+        }
+
+        params6 = {
+            "dist": "TDistribution",
+            "natural_gradient": False,
```

Comment:
this one seems to be the failing example. Any ideas?
I cannot spot any issue. Is this perhaps a bug in `ngboost` itself?

Reply:
t-distribution does have non-degenerate natrual gradient afaik, so that might not be it?
But perhaps it's worth trying a different combination here, e.g., normal with natural gradient, and tdistibution plain.

Reply:
I tried running with the `ngboost`'s regressor itself it still gave `Singular Matrix` error for the `load_diabetes` dataset for `T-Distribution`. For all other distributions it works as expected. So just in case it was actually due to some mathematical working involving the determinant of the matrix to not be equal to 0 in case of T-Distribution, I tried with `load_breast_cancer` dataset ,even for this it is giving Singular Matrix error. So it might actually be the case that there's a bug in `ngboost` T-Distribution itself.

Reply:
what we could do, remove the parameter set in this PR, but file a bug report on `sktime`, with code that uses these parameters. Unless of course there is another explanation or a fix.